src/lib/SpeculationEngine.ts

```diff
@@ -0,0 +1,117 @@
+import { LandRequest, LandRequestStatus } from '../db';
+import { Logger } from './Logger';
+import { StateService } from './StateService';
+
+/**
+ *  SpeculationEngine uses impact of a PR to reorder it while placing the PRs on the running slots.
+ *  The lower impact PRs are given higher precedence.
+ *  Impact meta data is processed and send to landkid by the consuming repo using build statuses.
+ * */
```

Comment:
Nit: Might be worth explaining why this is beneficial. I.e. to reduce the likelihood of a landkid failures due to dependencies